src/modules/package.rs

```diff
@@ -0,0 +1,108 @@
+use super::Segment;
+use crate::context::Context;
+use ansi_term::Color;
+use serde_json;
+use std::fs::File;
+use std::io;
+use std::io::Read;
+use std::path::PathBuf;
+use toml;
+
+/// Creates a segment with the current package version
+///
+/// Will display if a version is defined for your Node.js or Rust project (if one exists)
+pub fn segment(context: &Context) -> Option<Segment> {
+    match get_package_version(context) {
+        Some(package_version) => {
+            const PACKAGE_CHAR: &str = "📦";
+            const SEGMENT_COLOR: Color = Color::Red;
+
+            // TODO: Make the prefix for the module "is "
+            let mut segment = Segment::new("package");
+            segment.set_style(SEGMENT_COLOR.bold());
+
+            segment.set_value(format!("{} {}", PACKAGE_CHAR, package_version));
+
+            Some(segment)
+        }
+        None => None,
+    }
+}
+
+// TODO: Combine into one function and just call for different file names!
+fn is_cargo_toml(dir_entry: &PathBuf) -> bool {
+    dir_entry.is_file() && dir_entry.file_name().unwrap_or_default() == "Cargo.toml"
+}
+
+fn is_package_json(dir_entry: &PathBuf) -> bool {
+    dir_entry.is_file() && dir_entry.file_name().unwrap_or_default() == "package.json"
+}
+
+// TODO: Move to `utils.rs` file and import
+fn read_file(file_name: &str) -> io::Result<String> {
+    let mut file = File::open(file_name)?;
+    let mut data = String::new();
+
+    file.read_to_string(&mut data)?;
+
+    Ok(data)
+}
+
+fn extract_cargo_version(file_contents: String) -> Option<String> {
+    let cargo_toml = file_contents.parse::<toml::Value>().ok()?;
+
+    match cargo_toml["package"]["version"].as_str() {
+        Some(raw_version) => {
+            let version = format_version(raw_version.to_string());
+            Some(version)
+        }
+        None => None,
+    }
+}
+
+fn extract_package_version(file_contents: String) -> Option<String> {
+    let json: Option<serde_json::Value> = serde_json::from_str(&file_contents).ok()?;
+
+    match json {
+        None => None,
```

Comment:
nitpick: I would put `None` at the end of the match statement to match the other matches.